config/types/etcd.go

```diff
@@ -32,7 +33,9 @@ var (
 )
 
 // Options can be the options for any Etcd version
-type Options interface{}
+type Options interface {
```

Comment:
But that's how Go works. All of this is determined at runtime instead of at compile time. You don't specify which interfaces you are implementing (e.g. Rust's `impl`), it uses duck-typing to figure out which interfaces are implicitly satisfied. Maybe that is broken across packages.